p256/src/arithmetic/scalar.rs

```diff
@@ -366,21 +326,20 @@ impl Scalar {
 
     /// Returns the SEC1 encoding of this scalar.
     pub fn to_bytes(&self) -> FieldBytes {
-        let mut ret = FieldBytes::default();
-        ret[0..8].copy_from_slice(&self.0[3].to_be_bytes());
-        ret[8..16].copy_from_slice(&self.0[2].to_be_bytes());
-        ret[16..24].copy_from_slice(&self.0[1].to_be_bytes());
-        ret[24..32].copy_from_slice(&self.0[0].to_be_bytes());
-        ret
+        self.0.to_be_byte_array()
     }
 
     /// Returns self + rhs mod n
+    // TODO(tarcieri): use `UInt::add_mod`
```

Comment:
@dignifiedquire originally as part of this PR I did attempt to use `add_mod` and ran into some test failures. I'll open a draft followup with that change after this gets merged